test/test_pymbolic.py

```diff
@@ -938,6 +938,13 @@ def test_cached_mapper_differentiates_float_int():
 # }}}
 
 
+def test_nodecount():
+    expr = prim.Sum((4, 4.0))
```

Comment:
Test the corner cases. (repeated nodes, CSEs, etc.)

Reply:
Done in 89e2777